plugins/catalog-backend/src/service/NextEntitiesCatalog.ts

```diff
@@ -150,7 +165,7 @@ function parseFilter(
     let cumulativeQuery = query;
     for (const subFilter of filter.allOf ?? []) {
       cumulativeQuery = cumulativeQuery.andWhere(subQuery =>
-        parseFilter(subFilter, subQuery, db),
+        parseFilter(subFilter, subQuery, db, negate),
```

Comment:
Hmm - I don't believe this is correct. If I'm reading the code correctly, it would lead to taking a filter like `{ not: { allOf: [ A, B, C ] } }` and expressing it as `{ allOf: [ !A, !B, !C ] }`, which is not the same thing. I think we probably need to either explore making use of something like [`whereNot`](https://knexjs.org/#Builder-whereNot) to keep the negations at the correct level. If that doesn't work, I think there's a bunch of legwork to do to flip logical ANDs and ORs when negating filters, since `!(A && B)` == `!A || !B`.

Reply:
Thanks for the catch, whereNot seems to work!